### ISSUE TITLE
Prefix for @Component

### ISSUE THREAD

**Issue by SebastianStehle** [AngularClass/angular2-conventions-loader#5]
Hi,

I would like to have another feature ;) ... I am too lazy to write imports for all types, therefore I just use 

```
import * as Ng2 from '@angular/core';
```

Can you add something like a prefix for @Component and @Directive?

**Comment by gdi2290** [comment]
are you saying you want to type a class that have Component in the name and have it auto include the decorator? so

```
export class AppComponent  {
}
```
with have
```
@Component({})
export class AppComponent  {
}
```

or auto inject imports?
so
```
@Component({})
export class AppComponent  {
}
```
with have 
```
import { Component } from '@angular/core';
@Component({})
export class AppComponent  {
}
```

**Comment by SebastianStehle** [comment]
No, sorry.

I am too lazy to include all types in the import (especially for Components where you have OnInit, OnDestroy, Changes, ElementRef, Renderer ...)

Therefore I just use:

     import * as Ng2 from '@angular/core';

     @Ng2.Component({ ... });
     export class MyComponent {
     }

I tried to integrate it yesterday. The idea was to add an ngAlias parameter and then to do some replaces before the Line 87

     var ngAlias = '';
     if (query.ngAlias && typeof query.ngAlias === 'string') {
         ngAlias = query.ngAlias + '.';
     }
     if (ngAlias) {
          source = source.replace(new RegExp('@' + ngAlias + 'Component', '@Component');
          source = source.replace(new RegExp('@' + ngAlias + 'Directive', '@Directive');
     }

     and change Line 166 to

     return @' + ngAlias + decorator + '({' + metadata + '})';

**Comment by awerlang** [comment]
Would an IDE be able to deal with this? How would it recognized where Component() is from?

**Comment by SebastianStehle** [comment]
The IDE?

**Comment by awerlang** [comment]
I mean this:

![image](https://cloud.githubusercontent.com/assets/589286/17372882/19aa39c2-597c-11e6-938c-55d525befdd1.png)

**Comment by SebastianStehle** [comment]
You mean it can generate the imports? Probably, but VS Code cannot, or at least not with a single click ;)

**Comment by awerlang** [comment]
Yes, IDEs are meant to help with this. WebStorm can do it, and VS Code maybe someday.

What I meant is if you omit this prefix on your source code and having a plugin injecting it on transpilation, you'll lose all the goodness in the IDE e.g. intellisense, type checking, etc. Unless the IDE implement this convention, which is unlikely.

**Comment by SebastianStehle** [comment]
I dont understand what you mean. I use this prefix/alias because I dont like long import statements, e.g.

     import { Component, Input, Output, OnChanges, OnInit, OnDestroy } from '@angular/core';

You can use any alias you want, it is valid typescript and you get intellisense for it. But this package here, does not recognize aliases, because it tries to find '@Component' in the code. I am confused at the moment :D

**Comment by SebastianStehle** [comment]
@gdi2290 are you going to add this feature?

**Comment by michaeljota** [comment]
To autoimport (and other goods) try [Typescript Hero](https://github.com/buehler/typescript-hero)

@SebastianStehle did you try Typescript Hero? Did it work as you expected? I know everyone it's free to work as they think it's better, but destructing imports have it's benefits, even more when you are working with Webpack and a tree shaking-like feature.

**Comment by SebastianStehle** [comment]
Yes, I migrated my code. The tree shaking might not work correctly but i have seen no difference in the size of the bundle.
